docs/website/contents/for-developers/QueryVersioning.md

```diff
@@ -0,0 +1,77 @@
+
+   The mapping from `NodeToClientVersion`s to `CardanoNodeToClientVersion`s is [`supportedNodeToClientVersions`][supportedNodeToClientVersions]
+
+Additionally, all versions larger than a certain `NodeToClientVersion` (see [`latestReleasedNodeVersion`][latestReleasedNodeVersion]) are considered experimental, and are only offered in the initial version negotiation when a flag (currently, `ExperimentalProtocolsEnabled`) is set. Also see [`limitToLatestReleasedVersion`][limitToLatestReleasedVersion] and its call/usage sites.
+
+### Checks
+
+The supported query versions are only enforced in the [Shelley query encoder][shelley-encodeNodeToClient], ie in code run by clients. The server will answer queries even if the negotiated version is older than expected (which might be the case with a custom client implementation), following the robustness principle.
+
+## How to version a new query
+
+ 1. Determine whether the query is supposed to be experimental, ie only used for demo purposes/specific to an unreleased era.
+
+ 2. Check whether you need a new [`NodeToClientVersion`][n2c].
+
+     - If the query is experimental, you only need one if there does not already exist a [`NodeToClientVersion`][n2c] beyond the [`latestReleasedNodeVersion`][latestReleasedNodeVersion] (usually, it should already exist).
+
+     - If the query is not experimental, you need one if the current [`latestReleasedNodeVersion`][latestReleasedNodeVersion] is already used in a released version of the node. For this, check the version of `ouroboros-consensus-cardano` in the latest node release, and navigate to the corresponding [`latestReleasedNodeVersion`][latestReleasedNodeVersion].
+
+    If positive, create a corresponding PR in the [Network repository][network-repo], and wait for a new release to CHaP.
+
+ 3. Depending on the previous step:
+
+     - if necessary, add a new [`CardanoNodeToClientVersion`][cardano-n2c] and adapt [`supportedNodeToClientVersions`][supportedNodeToClientVersions] (as well as Shelley's [`supportedNodeToClientVersions`][shelley-supportedNodeToClientVersions], which is only used in tests)
+
+     - if necessary, add a new [`ShelleyNodeToClientVersion`][shelley-n2c] and adapt [`querySupportedVersion`][query-supported-version]
+
+ 4. In many cases, these changes will make our golden tests fail (due to missing files); to fix CI, run the tests locally and check in the newly generated files.
+
+[n2c]: https://input-output-hk.github.io/ouroboros-network/ouroboros-network/Ouroboros-Network-NodeToClient.html#t:NodeToClientVersion
+[network-spec]: https://input-output-hk.github.io/ouroboros-network/pdfs/network-spec/network-spec.pdf
+[shelley-n2c]: https://github.com/input-output-hk/ouroboros-consensus/blob/35e444f1440cef34e0989519f025231241397674/ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/NetworkProtocolVersion.hs#L17
```

Comment:
Yeah, I am also not sure about this, there is a tradeoff between:

 - Code links are actually pointing to the correct thing. (satisfied by permalinks)
 - Code links are pointing you to the most recent version of the code. (not satisfied by permalinks)

> Perhaps we should simply link to the module file (without the line number), and refer to the identifier people should look for.

This would be one approach, another one would be to make a pass through the permalinks in our docs every ~3 months and update them?

Reply:
Maybe we should keep the permanent links and update them in an on-demand basis.